kore/src/Kore/IndexedModule/IndexedModule.hs

```diff
@@ -111,17 +117,30 @@ module data in raw-ish form.
 All 'IndexedModule' instances should be returned by
 'indexedModuleWithDefaultImports'.
 -}
-
--- TODO (thomas.tuegel): Consider splitting IndexedModule into separate sort,
--- symbol, and axiom indices.
-data IndexedModule pat declAtts axiomAtts = IndexedModule
+data IndexedModuleSyntax pat declAtts = IndexedModuleSyntax
     { indexedModuleName :: !ModuleName
     , indexedModuleAliasSentences ::
         !(Map.Map Id (declAtts, SentenceAlias pat))
     , indexedModuleSymbolSentences ::
         !(Map.Map Id (declAtts, SentenceSymbol))
     , indexedModuleSortDescriptions ::
         !(Map.Map Id (Attribute.Sort, SentenceSort))
+    , indexedModuleImportsSyntax ::
+        ![ ( declAtts
+           , Attributes
+           , IndexedModuleSyntax pat declAtts
+           )
```

Comment:
I'd rather define a separate product type with record syntax here instead of the triple. Not sure if this has an impact here, but the `(,,)` is lazy, and you might or you might not want this behavior. Defining your own type with record syntax will improve both readability and will let you make it as lazy as you want.

Another thing to note is that the list is also lazy, but again I'm not sure if this has any impact on your code or not.

Reply:
So, this is proving quite tricky to get to work. The current code relies on things like `Lens._1` and `Lens._3`, and when I try to replace this with a `field`, it reports some very strange errors involving "no instance for GProductLens". I'm not sure it's really worth it, especially considering that this tuple structure is already in use in this file in other fields involving imports.

Reply:
I can keep looking into it if you really think it's necessary, but it's going to be a pretty involved change.